core/dbt/cli/params.py

```diff
@@ -6,6 +6,12 @@
 from dbt.cli.resolvers import default_project_dir, default_profiles_dir
 from dbt.version import get_version_information
 
+add_package = click.option(
+    "--add",
+    help="Add a package to current package spec with `--package` and `--package-version`",
```

Comment:
is --source also only valid in the context of `--add`?

Reply:
I don't _love_ the location of the validation here (all the way in task running), but it's where we're doing a similar check for the dry-run flag: https://github.com/dbt-labs/dbt-core/pull/8408/files#diff-77c52d1632e496415c4b241f4eb7ced5e8f7c6cbe207955177b2e8c6c6a54227R217-R220

I think it's tricky to do this validation in click.

Some alternative locations for doing this validation would be either: 
1. in the `deps` click command method itself in main.py, or
2.  implementing something similar to `Flags._assert_mutually_exclusive` (perhaps `assert_dependency`- keeping click/arg parsing concerns happening at a higher level. 

I think ideally we could leverage click to do this validation by structuring `dbt deps` as a group and `dbt deps add` as a subcommand, but I understand we're not going down that path for backwards compatibility reasons - which is fair! So I'd bias to doing the simpler thing if anything (option 1) - to avoid legitimizing this pattern much, favoring subcommands when possible.